SYCL/InvokeSimd/call_vadd_1d_loop.cpp

```diff
@@ -0,0 +1,236 @@
+          << filter_string << "' does not contain such substrings.\n";
+      return -1;
+    }
+    // If "SYCL_DEVICE_FILTER" not defined, only allow gpu device
+    return device.is_gpu() ? 1000 : -1;
+  }
+};
+
+inline auto createExceptionHandler() {
+  return [](exception_list l) {
+    for (auto ep : l) {
+      try {
+        std::rethrow_exception(ep);
+      } catch (sycl::exception &e0) {
+        std::cout << "sycl::exception: " << e0.what() << std::endl;
+      } catch (std::exception &e) {
+        std::cout << "std::exception: " << e.what() << std::endl;
+      } catch (...) {
+        std::cout << "generic exception\n";
+      }
+    }
+  };
+}
+
+#ifndef INVOKE_SIMD
+#define INVOKE_SIMD 1
+#endif
+
+constexpr bool use_invoke_simd = INVOKE_SIMD != 0;
+
+int main(void) {
+  constexpr unsigned Size = 1024;
+  constexpr unsigned GroupSize = 4 * VL;
+
+  queue q(ESIMDSelector{}, createExceptionHandler());
+
+  auto dev = q.get_device();
+  std::cout << "Running on " << dev.get_info<info::device::name>() << "\n";
+  auto ctxt = q.get_context();
+  // TODO: release memory in the end of the test
+  float *A =
+      static_cast<float *>(malloc_shared(Size * sizeof(float), dev, ctxt));
+  float *B =
+      static_cast<float *>(malloc_shared(Size * sizeof(float), dev, ctxt));
+  float *C =
+      static_cast<float *>(malloc_shared(Size * sizeof(float), dev, ctxt));
+
+  for (unsigned i = 0; i < Size; ++i) {
+    A[i] = B[i] = i;
+    C[i] = -1;
+  }
+
+  sycl::range<1> GlobalRange{Size / VL};
+  // Number of workitems in each workgroup.
+  sycl::range<1> LocalRange{GroupSize};
+
+  sycl::nd_range<1> Range(GlobalRange, LocalRange);
+
+  try {
+    auto e = q.submit([&](handler &cgh) {
+      cgh.parallel_for<class Test>(
+          Range, [=](nd_item<1> ndi) [[intel::reqd_sub_group_size(VL)]] {
+            sub_group sg = ndi.get_sub_group();
+            group<1> g = ndi.get_group();
+            uint32_t i = sg.get_group_linear_id() * VL +
+                         g.get_group_linear_id() * GroupSize;
+            int wi_id = i + sg.get_local_id();
+            // uint32_t wi_id = i + sg.get_local_id();
+            // This simpler offset computation also works,
+            // but using the original one above for
+            // consistency.
+            // int wi_id = ndi.get_global_linear_id();
+
+            if constexpr (use_invoke_simd) {
+              float res = invoke_simd(sg, SIMD_CALLEE, uniform{A}, uniform{B},
+                                      uniform{C}, wi_id);
+              // CHECK: %{{[A-z0-9_.]*}} = call x86_regcallcc float @_Z33__regcall3____builtin_invoke_simd{{[A-z0-9_.]*}}(<16 x float> (float addrspace(4)*, float addrspace(4)*, float addrspace(4)*, <16 x i32>)* noundef nonnull @{{[A-z0-9_.]*}}spmd2simd{{[A-z0-9_.]*}}, float addrspace(4)* noundef %{{[A-z0-9_.]*}}, float addrspace(4)* noundef %{{[A-z0-9_.]*}}, float addrspace(4)* noundef %{{[A-z0-9_.]*}}, i32 noundef %{{[A-z0-9_.]*}}) [[ATTRS1:#[0-9]+]]
```

Comment:
E2E tests should not check IR. This is already done by unit tests. So please remove from here (some of it may go to llvm/test/tools/sycl-post-link or similar if you feel it adds coverage).